src/websocket/webSocketClient.ts

```diff
@@ -0,0 +1,131 @@
+import { Api } from "coder/site/src/api/api";
+import {
+	GetInboxNotificationResponse,
+	ProvisionerJobLog,
+	ServerSentEvent,
+	Workspace,
+	WorkspaceAgent,
+} from "coder/site/src/api/typesGenerated";
+import { ProxyAgent } from "proxy-agent";
+import { ClientOptions } from "ws";
+import { coderSessionTokenHeader } from "../api";
+import { WsLogger } from "../logging/netLog";
+import { Storage } from "../storage";
+import { OneWayCodeWebSocket } from "./oneWayCodeWebSocket";
+
+/**
+ * WebSocket client for Coder API connections.
+ *
+ * Automatically configures logging for WebSocket events:
+ * - Connection attempts and successful opens at the trace level
+ * - Connection closes at the trace level
+ * - Errors at the error level
+ */
+export class CoderWebSocketClient {
```

Comment:
Really nice how the web socket client is abstracting things, so much better than what we had. :+1: 

Just to make sure I am on the right page, these (or at least some of these) already exist in `coder/coder`, right?  But we have to duplicate them because they use browser web socket right?

At some point we may want to modify `coder/coder`'s `OneWayWebSocket` so we can pass in a web socket constructor and use it externally as well.